src/main/java/seedu/address/logic/commands/FilterCommand.java

```diff
@@ -27,9 +30,10 @@ public FilterCommand(DatePredicate predicate) {
     public CommandResult execute() throws CommandException {
 
         RouteOptimization route = new RouteOptimization();
+        List<String> optimizedRoute;
 
         model.updateFilteredPersonList(predicate);
-        route.getAddresses(model);
+        optimizedRoute = route.getAddresses(model);
```

Comment:
@ncaminh optimizedRoute is the list of addresses in the order that they should be mapped!!!